### PR TITLE
Fix build CI

### PR DESCRIPTION
**Pull request by fcollonval**
- Bump all npm dependencies  
  Got hit by a breaking API in `react-notifications-component` v3.3.0
- Switch to Python pathlib
- Use common helper for listing federated extensions
- Upgrade setup.py
- Add installation of quetz + quetz-frontend in the CI

### CODE REVIEW

File: .github/actions/base-setup/action.yml [hunk 1]
```diff
@@ -25,19 +25,19 @@ runs:
...
     - name: Install the server
-      shell: bash
+      shell: bash -l {0}
       run: |
-        pip install -e .
+        python -m pip install -e .
       working-directory: ${{ inputs.server_folder }}
 
     - name: Install the frontend
       shell: bash
```
Comment (wolfv) [inline .github/actions/base-setup/action.yml]: `bash -l {0}` may be missing here as well?
We can also define it as part of the defaults settings: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-set-the-default-shell-and-working-directory

Comment (fcollonval) [reply .github/actions/base-setup/action.yml]: Thanks for the tip. Apparently it is not supported within a custom action definition

**Comment by wolfv** [comment]
Awesome!

File: .github/actions/base-setup/action.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+runs:
+  using: 'composite'
+  steps:
+    - uses: actions/checkout@v2
+      with:
+        repository: mamba-org/quetz
+        ref: master
+        path: ${{ inputs.server_folder }}
+
+    - uses: conda-incubator/setup-miniconda@v2
```
Comment (wolfv) [inline .github/actions/base-setup/action.yml:21]: Do you think we could use our micromamba action here?

Comment (wolfv) [reply .github/actions/base-setup/action.yml:21]: https://github.com/mamba-org/provision-with-micromamba

**Comment by wolfv** [comment]
Happy to merge this. Do you want to squash commits a bit or shoudl we keep this history?

File: .github/actions/base-setup/action.yml (hunk 2 above)
Comment (fcollonval) [reply .github/actions/base-setup/action.yml]: Thanks for the suggestion.

Is it possible to set up package caching with that action?

**Comment by fcollonval** [comment]
> Happy to merge this. Do you want to squash commits a bit or shoudl we keep this history?

I usually like to apply _Squash and Merge_ instead of simple _Merge_ on my PR. So feel free to do so with this PR.

**Comment by wolfv** [comment]
Done! Thanks!

Some people don't like all commits squashed :)

**Pull request merged**